src/components/table/data-table.tsx

```diff
@@ -54,15 +58,41 @@ export function DataTable<TData, TValue>({
     manualPagination: pagination !== undefined,
     pageCount: pagination?.pageCount ?? 0,
     onPaginationChange: pagination?.onChange,
+    onRowSelectionChange: setRowSelection,
     state: {
       columnFilters,
       pagination: pagination?.state,
+      rowSelection,
     },
     manualFiltering: true,
   });
 
+  useEffect(() => {
+    if (onSelectionChange) {
+      const selectedRowsIndexes = Object.keys(rowSelection).map(Number);
+      const rows = table.getRowModel().rows;
+      const selectedRows = rows.filter((_, i) => {
+        return selectedRowsIndexes.includes(i);
+      });
+
+      onSelectionChange(selectedRows);
+    }
+  }, [rowSelection]);
+
+  useEffect(() => {
+    setRowSelection({});
+  }, [table.getRowModel().rows.length]);
+
+  const noOfSelectedRows = table.getFilteredSelectedRowModel().rows.length;
+
   return (
     <>
+      {noOfSelectedRows > 0 && (
```

Comment:
This shifts the layout when the first trace is selected. Suggestion: add `(n selected)` to the action dropdown